lib/index.js

```diff
@@ -215,13 +202,71 @@ function lowerCaseObjectKeys(obj) {
   return newObject;
 }
 
+function validateDate(date, key) {
+  // created & expires are not required
+  if(date === undefined || date === null) {
+    return date;
+  }
+  if(date instanceof Date) {
+    const timestamp = date.getTime() / 1000;
+    if(key === 'created') {
+      return Math.ceil(timestamp);
+    }
+    if(key === 'expires') {
+      return Math.floor(timestamp);
+    }
+    throw new TypeError(`Unknown date pseudo-header ${key}`);
+  }
+  if(Number.isInteger(date)) {
+    // we need to accept that the integer is a unix time stamp
+    return date;
+  }
+  if(typeof date === 'string') {
+    // we assume this is a unix time stamp too
+    const number = Number(date);
+    if(Number.isInteger(number)) {
+      return number;
+    }
+  }
+  throw new TypeError(
+    `"${key}" must be a UNIX timestamp or JavaScript Date.`);
+}
+
+/**
+ * Validates and formats date pseudo-headers.
+ *
+ * @param {Object} options - Options to use.
+ *
+ * @returns {Object} Formatted options.
+*/
+function validateDates(options) {
```

Comment:
I'm not sure we want these validation checks. Why do we have them? Also, we won't be able to easily do test vector checks with these, right?

Reply:
this has been changed to `validateDatesRange` and then the `validateDate` only ensures the date is a time stamp.